spec/models/activity_spec.rb

```diff
@@ -2204,6 +2204,55 @@ def create_report(organisation, fund)
     end
   end
 
+  describe "#ensure_linked_activity_reciprocity" do
+    context "when adding a linked activity" do
+      it "links the newly linked activity back to self" do
+        oda_activity = create(:project_activity, :ispf_funded, is_oda: true)
+        non_oda_activity = create(:project_activity, :ispf_funded,
+          is_oda: false,
+          linked_activity: oda_activity)
+
+        oda_activity.reload
+        expect(oda_activity.linked_activity).to eq(non_oda_activity)
+      end
+    end
+
+    context "when removing a linked activity" do
+      it "unlinks the old activity from self" do
```

Comment:
Do you mean reword the test description, or add a new test? I thought we covered all the combinations. Could you give us a code snippet or discuss synchronously to avoid extra heapings of confusion? 😅

Reply:
Ah, I think I understand what this means now - maybe rewording this to `it "ensures the link is removed from both sides of the relationship"` would have made more sense to me (I think I was confused by the use of `self` here.